app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/apps/TrackingProtectionAppsRepositoryTest.kt

```diff
@@ -0,0 +1,272 @@
+ */
+
+package com.duckduckgo.mobile.android.vpn.apps
+
+import android.content.pm.ApplicationInfo
+import android.content.pm.PackageManager
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import app.cash.turbine.test
+import com.duckduckgo.app.CoroutineTestRule
+import com.duckduckgo.mobile.android.vpn.feature.AppTpFeatureConfig
+import com.duckduckgo.mobile.android.vpn.feature.AppTpSetting
+import com.duckduckgo.mobile.android.vpn.trackers.FakeAppTrackerRepository
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+
+@RunWith(AndroidJUnit4::class)
+@OptIn(ExperimentalCoroutinesApi::class)
+class TrackingProtectionAppsRepositoryTest {
+
+    @get:Rule
+    var coroutineRule = CoroutineTestRule()
+
+    private val packageManager: PackageManager = mock()
+    private val appTrackerRepository = FakeAppTrackerRepository()
+    private val appTpFeatureConfig: AppTpFeatureConfig = mock()
+
+    private lateinit var trackingProtectionAppsRepository: TrackingProtectionAppsRepository
+
+    @Before
+    fun setup() {
+        whenever(appTpFeatureConfig.isEnabled(AppTpSetting.ProtectGames)).thenReturn(false)
+        whenever(packageManager.getInstalledApplications(PackageManager.GET_META_DATA)).thenReturn(INSTALLED_APPS.asApplicationInfo())
+        whenever(packageManager.getApplicationLabel(any())).thenReturn("App Name")
+        appTrackerRepository.appExclusionList = EXCLUSION_LIST.toSet()
+        appTrackerRepository.manualExclusionList = MANUAL_EXCLUSION_LIST.toMutableMap()
+        appTrackerRepository.systemAppOverrides = SYSTEM_OVERRIDE_LIST.toSet()
+
+        trackingProtectionAppsRepository =
+            RealTrackingProtectionAppsRepository(
+                packageManager, appTrackerRepository, coroutineRule.testDispatcherProvider, appTpFeatureConfig
+            )
+    }
+
+    @Test
+    fun whenGetExclusionAppListThenReturnExclusionList() = runTest {
+        val exclusionList = trackingProtectionAppsRepository.getExclusionAppsList()
+
+        assertEquals(
+            listOf("com.example.app2", "com.example.app3", "com.example.app5", "com.example.game", "com.example.system", "com.duckduckgo.mobile"),
+            exclusionList
+        )
+    }
+
+    @Test
+    fun whenIsProtectionEnabledCalledOnDisabledAppThenReturnFalse() = runTest {
+        whenever(packageManager.getApplicationInfo("com.example.app2", 0))
+            .thenReturn(ApplicationInfo().apply { packageName = "com.example.app2" })
+
+        val isEnabled = trackingProtectionAppsRepository.isAppProtectionEnabled("com.example.app2")
+
+        assertFalse(isEnabled)
+    }
+
+    @Test
+    fun whenIsProtectionEnabledCalledOnEnabledAppThenReturnFalse() = runTest {
```

Comment:
I think this should be `whenIsProtectionEnabledCalledOnEnabledAppThenReturnTrue`.